parser-typechecker/src/Unison/Util/Bytes.hs

```diff
@@ -86,6 +89,159 @@ at i bs = case Unison.Util.Bytes.drop i bs of
   Bytes (T.viewl -> hd T.:< _) -> Just (B.index hd 0)
   _ -> Nothing
 
+decodeNat64be :: Bytes -> Maybe (Word64, Bytes)
+decodeNat64be bs = do
+  b1 <- at 0 bs
+  b2 <- at 1 bs
+  b3 <- at 2 bs
+  b4 <- at 3 bs
+  b5 <- at 4 bs
+  b6 <- at 5 bs
+  b7 <- at 6 bs
+  b8 <- at 7 bs
+
+  let b = ((shiftL (fromIntegral b1) 56) .|. (shiftL (fromIntegral b2) 48) .|. (shiftL (fromIntegral b3) 40) .|. (shiftL (fromIntegral b4) 32) .|. (shiftL (fromIntegral b5) 24) .|. (shiftL (fromIntegral b6) 16) .|. (shiftL (fromIntegral b7) 8) .|. (fromIntegral b8))
+  return (b, (Unison.Util.Bytes.drop 8 bs))
+
+decodeNat64le :: Bytes -> Maybe (Word64, Bytes)
+decodeNat64le bs = do
+  b1 <- at 0 bs
+  b2 <- at 1 bs
+  b3 <- at 2 bs
+  b4 <- at 3 bs
+  b5 <- at 4 bs
+  b6 <- at 5 bs
+  b7 <- at 6 bs
+  b8 <- at 7 bs
+
+  let b = ((shiftL (fromIntegral b8) 56) .|. (shiftL (fromIntegral b7) 48) .|. (shiftL (fromIntegral b6) 40) .|. (shiftL (fromIntegral b5) 32) .|. (shiftL (fromIntegral b4) 24) .|. (shiftL (fromIntegral b3) 16) .|. (shiftL (fromIntegral b2) 8) .|. (fromIntegral b1))
+  return (b, (Unison.Util.Bytes.drop 8 bs))
+
+
+decodeNat32be :: Bytes -> Maybe (Word64, Bytes)
+decodeNat32be bs = do
+  b1 <- at 0 bs
+  b2 <- at 1 bs
+  b3 <- at 2 bs
+  b4 <- at 3 bs
+
+  let b = (shiftL (fromIntegral b1) 24) .|. (shiftL (fromIntegral b2) 16) .|. (shiftL (fromIntegral b3) 8) .|. (fromIntegral b4)
+  return (b, (Unison.Util.Bytes.drop 4 bs))
+
+decodeNat32le :: Bytes -> Maybe (Word64, Bytes)
+decodeNat32le bs = do
+  b1 <- at 0 bs
+  b2 <- at 1 bs
+  b3 <- at 2 bs
+  b4 <- at 3 bs
+
+  let b = (shiftL (fromIntegral b1) 24) .|. (shiftL (fromIntegral b2) 16) .|. (shiftL (fromIntegral b3) 8) .|. (fromIntegral b4)
+  return (b, (Unison.Util.Bytes.drop 4 bs))
+
+decodeNat16be :: Bytes -> Maybe (Word64, Bytes)
+decodeNat16be bs = do
+  b1 <- at 0 bs
+  b2 <- at 1 bs
+
+  let b = (shiftL (fromIntegral b1) 8) .|. (fromIntegral b2)
+  return (b, (Unison.Util.Bytes.drop 2 bs))
+
+decodeNat16le :: Bytes -> Maybe (Word64, Bytes)
+decodeNat16le bs = do
+  b1 <- at 0 bs
+  b2 <- at 1 bs
+
+  let b = (shiftL (fromIntegral b2) 8) .|. (fromIntegral b1)
+  return (b, (Unison.Util.Bytes.drop 2 bs))
+
+encodeNat64be :: Word64 -> Bytes
+encodeNat64be n =
```

Comment:
I think instead of implementing this manually, it'd be better to use on of the binary encoding libraries we already depend on. E.G. we use bytes which has a bunch of endian-specific functions:

    https://hackage.haskell.org/package/bytes-0.17.1/docs/Data-Bytes-Put.html

And can spit out a `ByteString` afterward.

Reply:
@dolio so I think we are doing much better now in this respect?

Reply:
Yeah, that looks better. I just default to preferring something in a more widely used library if we're already using it. That way there's hopefully less question whether some fiddly detail is mistaken (because more people would have noticed by now).